### PR TITLE
Add mdx-bundler example

### PR DESCRIPTION
**Pull request by tsriram**
This PR adds an example for using [mdx-bundler](https://github.com/kentcdodds/mdx-bundler) with Next.js. This is based on [with-mdx-remote](https://github.com/vercel/next.js/tree/canary/examples/with-mdx-remote) example, and just changes the library to mdx-bundler.

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

File: examples/with-mdx-bundler/components/CustomLink.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Link from 'next/link'
+
+export default function CustomLink({ as, href, ...otherProps }) {
+  return (
+    <>
+      <Link as={as} href={href}>
```
Comment (leerob) [inline examples/with-mdx-bundler/components/CustomLink.js:6]: `as` is no longer needed - can just be actually route value for href instead!

File: examples/with-mdx-bundler/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "with-mdx-remote",
+  "version": "1.0.0",
```
Comment (leerob) [inline examples/with-mdx-bundler/package.json:3]:
```suggestion
  "private": "true",
```

Comment (tsriram) [reply examples/with-mdx-bundler/package.json]: Ah, the perils of copy-pasting 😄

**Comment by leerob** [comment]
Closing as stale – if you would like to continue this, please open a new PR and we can take a look. Thank you so much! 🙏

**Pull request closed without merging**